drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -378,6 +382,45 @@ static int rf2xx_filter(struct device *dev,
 	return -ENOTSUP;
 }
 
+#if !defined(CONFIG_IEEE802154_RAW_MODE) && \
+	defined(CONFIG_NET_L2_OPENTHREAD)
+
+#define ACK_FRAME_LEN 3
+#define ACK_FRAME_TYPE (2 << 0)
+#define ACK_FRAME_PENDING_BIT (1 << 4)
+
+static void rf2xx_handle_ack(struct rf2xx_context *ctx, u8_t seq_number)
+{
+	struct net_pkt *pkt;
+	u8_t ack_psdu[ACK_FRAME_LEN] = {ACK_FRAME_TYPE, 0, seq_number};
+
+	pkt = net_pkt_alloc_with_buffer(ctx->iface, ACK_FRAME_LEN,
+					    AF_UNSPEC, 0, K_NO_WAIT);
+	if (!pkt) {
+		LOG_ERR("No free packet available.");
+		return;
+	}
+
+	/* Re-create ACK frame. */
+	if (ctx->trx_trac == RF2XX_TRX_PHY_STATE_TRAC_SUCCESS_DATA_PENDING) {
+		ack_psdu[0] |=  ACK_FRAME_PENDING_BIT;
+	}
+
+	memcpy(pkt->buffer->data, ack_psdu, ACK_FRAME_LEN);
```

Comment:
same as for pkt, allocate the buffer on stack and pre-fill it, no need to memcpy then.